tests/p2p/test_get_best_blockchain.py

```diff
@@ -0,0 +1,357 @@
+        self.assertIn(settings.CAPABILITY_GET_BEST_BLOCKCHAIN, protocol1.capabilities)
+        self.assertIn(settings.CAPABILITY_GET_BEST_BLOCKCHAIN, protocol2.capabilities)
+
+        # assert the protocol is in ReadyState
+        state1 = protocol1.state
+        state2 = protocol2.state
+        self.assertIsInstance(state1, ReadyState)
+        self.assertIsInstance(state2, ReadyState)
+
+        # assert ReadyState commands
+        self.assertIn(ProtocolMessages.GET_BEST_BLOCKCHAIN, state1.cmd_map)
+        self.assertIn(ProtocolMessages.BEST_BLOCKCHAIN, state1.cmd_map)
+        self.assertIn(ProtocolMessages.GET_BEST_BLOCKCHAIN, state2.cmd_map)
+        self.assertIn(ProtocolMessages.BEST_BLOCKCHAIN, state2.cmd_map)
+
+        # assert best blockchain contains the genesis block
+        self.assertIsNotNone(state1.best_blockchain)
+        self.assertIsNotNone(state2.best_blockchain)
+
+        # mine 100 blocks
+        miner = self.simulator.create_miner(manager1, hashpower=1e6)
+        miner.start()
+        trigger = StopAfterNMinedBlocks(miner, quantity=100)
+        self.assertTrue(self.simulator.run(7200, trigger=trigger))
+        miner.stop()
+
+        # assert best blockchain exchange
+        state1.send_get_best_blockchain()
+        state2.send_get_best_blockchain()
+        self.simulator.run(60)
+        self.assertEqual(settings.DEFAULT_BEST_BLOCKCHAIN_BLOCKS, len(state1.best_blockchain))
+        self.assertEqual(settings.DEFAULT_BEST_BLOCKCHAIN_BLOCKS, len(state2.best_blockchain))
+
+        self.assertIsInstance(state1.best_blockchain[0], BlockInfo)
+        self.assertIsInstance(state2.best_blockchain[0], BlockInfo)
+
+    def test_handle_get_best_blockchain(self):
+        manager1 = self.create_peer()
+        manager2 = self.create_peer()
+        conn12 = FakeConnection(manager1, manager2, latency=0.05)
+        self.simulator.add_connection(conn12)
+
+        # mine 100 blocks
+        miner = self.simulator.create_miner(manager1, hashpower=1e6)
+        miner.start()
+        trigger = StopAfterNMinedBlocks(miner, quantity=100)
+        self.assertTrue(self.simulator.run(7200, trigger=trigger))
+        miner.stop()
+
+        connected_peers1 = list(manager1.connections.connected_peers.values())
+        self.assertEqual(1, len(connected_peers1))
+        protocol2 = connected_peers1[0]
+        state2 = protocol2.state
+        self.assertIsInstance(state2, ReadyState)
+
+        connected_peers2 = list(manager2.connections.connected_peers.values())
+        self.assertEqual(1, len(connected_peers2))
+        protocol1 = connected_peers2[0]
+        state1 = protocol1.state
+        self.assertIsInstance(state1, ReadyState)
+
+        # assert compliance with N blocks inside the boundaries
+        state1.send_get_best_blockchain(n_blocks='1')
+        self.simulator.run(60)
+        self.assertFalse(conn12.tr1.disconnecting)
+
+        state2.send_get_best_blockchain(n_blocks='100')
```

Comment:
`'100'`? There are other parts of this test that are still using `str` instead of `int`.

Reply:
Done